cmd/utils/displayer/json.go

```diff
@@ -43,14 +43,17 @@ func newJSONDisplayer(stdout, stderr io.Reader) *jsonDisplayer {
 		stderrScanner = bufio.NewScanner(stderr)
 	}
 
+	commandContext, cancel := context.WithCancel(context.Background())
+
 	return &jsonDisplayer{
-		stdoutScanner: stdoutScanner,
-		stderrScanner: stderrScanner,
+		stdoutScanner:  stdoutScanner,
+		stderrScanner:  stderrScanner,
+		commandContext: commandContext,
```

Comment:
Adding the context inside a struct is something that shouldn't be done in Golang.

Contexts have a lifecycle, and they should be created and canceled appropriately. When you store a context inside a struct, you might lose track of when the context should be canceled, leading to potential resource leaks or incorrect behavior.

[More information here](https://go.dev/blog/context-and-structs)

Reply:
why do we need the context and cancel here?

Reply:
it was the way I found to reuse the context and cancel function in the needed places. I'm working on a refactor

Reply:
refactor to avoid store context in struct 687ca34461a0c5c27ec813884fb0c636d7baa01a

Reply:
do not check. There are bugs with the current proposal

Reply:
ready to review. As discussed offline, removes `ctx` and `cancel` func from struct requires a refactor in how executors and displayed are designed.